broker/broker-security-impl/src/main/java/org/codice/ddf/broker/security/impl/SecurityServerPluginImpl.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.broker.security.impl;
+
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.activemq.artemis.api.core.Message;
+import org.apache.activemq.artemis.core.server.ServerSession;
+import org.apache.activemq.artemis.core.transaction.Transaction;
+import org.apache.activemq.artemis.protocol.amqp.broker.AMQPMessage;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.wss4j.common.util.DOM2Writer;
+import org.codice.ddf.broker.security.api.SecurityServerPlugin;
+import org.codice.ddf.security.handler.api.UPAuthenticationToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SecurityServerPluginImpl implements SecurityServerPlugin {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SecurityServerPluginImpl.class);
+
+  private static final Map<String, Subject> SUBJECT_CACHE = new HashMap<>();
+
+  private static SecurityManager securityManager;
+
+  @Override
+  public void beforeSend(
+      ServerSession session,
+      Transaction tx,
+      Message message,
+      boolean direct,
+      boolean noAutoCreateQueue) {
+
+    LOGGER.trace(
+        String.format("Username: %s, Password: %s", session.getUsername(), session.getPassword()));
+    if (!SUBJECT_CACHE.containsKey(session.getUsername())) {
+      UPAuthenticationToken usernamePasswordToken =
+          new UPAuthenticationToken(session.getUsername(), session.getPassword());
+      try {
+        Subject subject = securityManager.getSubject(usernamePasswordToken);
+        SUBJECT_CACHE.put(session.getUsername(), subject);
+      } catch (SecurityServiceException e) {
+        LOGGER.warn("A bad security thing happened", e);
```

Comment:
❓Should we `return` at this point? If this exception is thrown, then `SUBJECT_CACHE.get(session.getUsername())` below will return null and an NPE will be thrown.

Reply:
Fixed